packages/connectors/src/metaMask.ts

```diff
@@ -14,32 +13,21 @@ import {
   RpcError,
   SwitchChainError,
   UserRejectedRequestError,
-  type WalletPermission,
   getAddress,
   numberToHex,
 } from 'viem'
 
 export type MetaMaskParameters = Evaluate<
-  ExactPartial<
-    Omit<
-      MetaMaskSDKOptions,
-      | 'checkInstallationImmediately'
-      | 'checkInstallationOnAllCalls'
-      | 'defaultReadOnlyChainId'
-      | 'readonlyRPCMap'
-    >
-  >
+  Omit<MetaMaskSDKOptions, '_source' | 'readonlyRPCMap'>
```

Comment:
Can we make `dappMetadata` optional and default to `{}`?  Since the following is valid.

```ts
const connectorFn = metaMask({
  dappMetadata: {},
})
```

That way you can get started with the connector faster:

```ts
const connectorFn = metaMask()
```